subsys/event_manager/event_manager_priv.h

```diff
@@ -182,25 +182,37 @@ extern "C" {
 
 
 #define _EVENT_TYPE_DECLARE(ename)					\
+	enum {_CONCAT(ename, _HAS_DYNDATA) = 0};			\
 	_EVENT_TYPE_DECLARE_COMMON(ename);				\
 	_EVENT_ALLOCATOR_FN(ename)
 
 
 #define _EVENT_TYPE_DYNDATA_DECLARE(ename)				\
+	enum {_CONCAT(ename, _HAS_DYNDATA) = 1};			\
 	_EVENT_TYPE_DECLARE_COMMON(ename);				\
 	_EVENT_ALLOCATOR_DYNDATA_FN(ename)
 
 
-#define _EVENT_TYPE_DEFINE(ename, init_log_en, log_fn, trace_data_pointer)		\
-	_EVENT_SUBSCRIBERS_ARRAY_TAGS(ename);						\
-	STRUCT_SECTION_ITERABLE(event_type, _CONCAT(__event_type_, ename)) = {		\
-		.name            = STRINGIFY(ename),					\
-		.subs_start      = _EVENT_SUBSCRIBERS_START_TAG(ename),			\
-		.subs_stop       = _EVENT_SUBSCRIBERS_END_TAG(ename),			\
-		.init_log_enable = init_log_en,						\
-		.log_event       = (IS_ENABLED(CONFIG_LOG) ? (log_fn) : (NULL)),	\
-		.trace_data      = (IS_ENABLED(CONFIG_EVENT_MANAGER_PROFILER_TRACER) ?	\
-					(trace_data_pointer) : (NULL)),			\
+#if IS_ENABLED(CONFIG_EVENT_MANAGER_PROVIDE_EVENT_SIZE)
+#define _EVENT_TYPE_DEFINE_SIZES(ename)             \
+	.struct_size = sizeof(struct ename),
+#else
+#define _EVENT_TYPE_DEFINE_SIZES(ename)
+#endif
+
+
+#define _EVENT_TYPE_DEFINE(ename, log_fn, trace_data_pointer, et_flags)				\
+	_EVENT_SUBSCRIBERS_ARRAY_TAGS(ename);							\
+	STRUCT_SECTION_ITERABLE(event_type, _CONCAT(__event_type_, ename)) = {			\
+		.name            = STRINGIFY(ename),						\
+		.subs_start      = _EVENT_SUBSCRIBERS_START_TAG(ename),				\
+		.subs_stop       = _EVENT_SUBSCRIBERS_END_TAG(ename),				\
+		.log_event       = (IS_ENABLED(CONFIG_LOG) ? (log_fn) : (NULL)),		\
+		.trace_data      = (IS_ENABLED(CONFIG_EVENT_MANAGER_PROFILER_TRACER) ?		\
+					(trace_data_pointer) : (NULL)),				\
+		.event_type_flags = ((_CONCAT(ename, _HAS_DYNDATA)) ?				\
+				(et_flags | BIT(EVENT_FLAGS_HAS_DYNDATA)) : (et_flags)),	\
```

Comment:
Maybe we should explicitly clear the flag too (so that we are sure that it always has proper value). Now we do not have a build assert that ensures user did not set this flag.